src/mlpack/core/data/imputer.hpp

```diff
@@ -0,0 +1,125 @@
+/**
+ * @file imputer.hpp
+ * @author Keon Kim
+ *
+ * Defines Imputer(), a utility function to replace missing variables
+ * in a dataset.
+ */
+#ifndef MLPACK_CORE_DATA_IMPUTER_HPP
+#define MLPACK_CORE_DATA_IMPUTER_HPP
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/data/impute_strategies/custom_strategy.hpp>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * This class implements a way to replace target values. It is dependent on the
+ * user defined Strategy and Mapper used to hold dataset's information.
+ *
+ * @tparam Option of imputation strategy.
+ * @tparam Mapper that is used to hold dataset information.
+ * @tparam primitive type of input and output's armadillo matrix.
+ */
+template<typename MatType, typename Mapper, typename Strategy>
+class Imputer
+{
+ private:
+  Strategy strat;
+ public:
+  Imputer()
```

Comment:
Maybe we could store the Mapper, Strategy and transpose information in the class, ask the users to pass them into the constructor.This way the Strategy could save some "states", which should help us to reduce some computation(like MeanStrategy)